src/chart/helper/whiskerBoxCommon.ts

```diff
@@ -52,6 +52,16 @@ class WhiskerBoxCommonMixin<Opts extends CommonOption> {
 
     defaultValueDimensions: CoordDimensionDefinition['dimsDef'];
 
+    /**
+     * @private
+     */
+    _hasEncodeRule(key: string) {
+        const encodeRules = this.getEncode();
+        if(encodeRules?.data?.has(key))
```

Comment:
We avoid using `?.` under compatible considerations. Please use `encodeRules && encodeRules.data && encodeRules.data.has(key)` instead.

Reply:
Thanks for the review @Ovilia
Changes pushed.